cloudformation/custom_resource/cluster-1-click.yaml

```diff
@@ -0,0 +1,98 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: >
+  AWS ParallelCluster CloudFormation Template
+
+Parameters:
+  KeyName:
+    Description: KeyPair to login to the head node
+    Type: AWS::EC2::KeyPair::KeyName
+
+  AvailabilityZone:
+    Description: Availability zone where instances will be launched
+    Type: AWS::EC2::AvailabilityZone::Name
+    Default: us-east-2a
+
+Mappings:
+  ParallelCluster:
+    Constants:
+      Version: 3.6.0
+      Bucket: '' # For debug purposes only
+
+Conditions:
+  UseCustomBucket: !Not [!Equals [!FindInMap [ParallelCluster, Constants, Bucket], '']]
+
+Resources:
+  PclusterClusterProvier:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      Parameters:
+        CustomBucket: !If [
+            UseCustomBucket,
+            !FindInMap [ParallelCluster, Constants, Bucket ],
+            !Ref AWS::NoValue
+          ]
+      TemplateURL: !Sub
+        - https://${Bucket}.s3.${AWS::Region}.amazonaws.com/parallelcluster/${Version}/templates/custom_resource/cluster.yaml
+        - { Version: !FindInMap [ParallelCluster, Constants, Version ],
+            Bucket: !If [
+              UseCustomBucket,
+              !FindInMap [ParallelCluster, Constants, Bucket ],
+              !Sub "${AWS::Region}-aws-parallelcluster"
+            ]
+          }
+
+  PclusterVpc:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      Parameters:
+        PublicCIDR: 10.0.0.0/24
+        PrivateCIDR: 10.0.16.0/20
+        AvailabilityZone: !Ref AvailabilityZone
+      TemplateURL: !Sub
+        - https://${Bucket}.s3.${AWS::Region}.amazonaws.com/parallelcluster/${Version}/templates/networking/public-private.cfn.json
+        - { Version: !FindInMap [ParallelCluster, Constants, Version ],
+            Bucket: !If [
+              UseCustomBucket,
+              !FindInMap [ParallelCluster, Constants, Bucket ],
+              !Sub "${AWS::Region}-aws-parallelcluster"
+            ]
+          }
+
+  PclusterCluster:
+    Type: Custom::PclusterCluster
+    Properties:
+      ServiceToken: !GetAtt [ PclusterClusterProvider , Outputs.ServiceToken ]
+      ClusterName: !Sub 'c-${AWS::StackName}'
+      ClusterConfiguration:
+        DevSettings: !If
+          - UseCustomBucket
+          -
+            AmiSearchFilters:
+              Owner: self
+          - !Ref AWS::NoValue
```

Comment:
I believe the NoValue here makes it not include the DevSettings, however I have checked (during debugging) that if this isn't included the YAML is still valid (but it will fail to find an official AMI)